SD-2853: add docs on how to use XFER (#536)
SparkPost/support-docs: momentum/4/outbound-xfer.md

---
title: "Configuring XFER for Outbound Mail Delivery"
description: "By default Momentum will attempt to deliver mail via the SMTP protocol to the destination MX server.  The XFER delivery method can be used to transfer mail over HTTP to another Momentum server that will handle delivery."
---

## <a name="outbound_mail.XFER"></a> XFER

By default Momentum will attempt to deliver mail via the SMTP protocol to the destination MX server.  The XFER delivery method can be used to transfer mail over HTTP to another Momentum server that will handle delivery.  Some example use cases where this could be useful include transferring tempfailed messages to another server for handling retries, and transferring mail to specific domains to another server that is configured with different settings.

> To receive XFER messages from another server configuration like the following is needed:

Reviewer: subheading here for Receiving Server Setup?
```suggestion
### Server Config for Receiving XFer Messages
To receive XFER messages from another server configuration like the following is needed:
```

> To receive XFER messages from another server configuration like the following is needed:

Reviewer 2: Nitpick
```suggestion
To receive XFER messages from another server, configuration like the following is needed:
```

```
scriptlet "scriptlet" {
  # load the XFER server
  script xfer_api {
    source = "msys.xfer.server"
  }
}

# stand up an HTTP listener on port 8080
HTTP_Listener {
  Listen ":8080" {}
}
```

> To reroute delayed messages to an XFER server configuration like the following is needed (assuming the XFER server is running on 10.10.10.10):

Reviewer: ```suggestion
### Sending Server for XFer Delayed Messages
To reroute delayed messages to an XFER server configuration like the following is needed (assuming the XFER server is running on 10.10.10.10):
```

> To reroute delayed messages to an XFER server configuration like the following is needed (assuming the XFER server is running on 10.10.10.10):

Reviewer 2: Aanother nitpick
```suggestion
To reroute delayed messages to an XFER server, configuration like the following is needed (assuming the XFER server is running on 10.10.10.10):
```

```
# reroute the delayed message to the synthetic xfer.test domain
delayed_reroute_domain = "xfer.test"
Domain xfer.test {
  reroute_domain = "disabled"
  delayed_reroute_domain = "disabled"
  # use the XFER delivery method
  Delivery_Method = "delay_xfer"
  Delivery_Response_Timeout = "10000"  # 10s
  # XFER to 10.10.10.10
  Routes = (
    "xfer://10.10.10.10?port=8080"
  )
  HTTP_URI = "/delay_message"
  HTTP_METHOD = "POST"
  HTTP_HOST = "localhost"
}
```

> To reroute messages for speicifc domains to an XFER server configuration like the following is needed:

Reviewer: speling

> To reroute messages for speicifc domains to an XFER server configuration like the following is needed:

Reviewer: ```suggestion
### Sending Server Config for Specific Domains
To reroute messages for specific domains to an XFER server configuration like the following is needed:
```

> To reroute messages for speicifc domains to an XFER server configuration like the following is needed:

Reviewer 2: Typo fix
```suggestion
To reroute messages for specifc domains to an XFER server, configuration like the following is needed:
```
